Overview.src.html

```diff
@@ -3216,6 +3266,59 @@ <h4 id="should-response-to-request-be-blocked-due-to-nosniff?"><dfn title="shoul
+          when it sets up a TLS connection to that server, creating a Token Binding Id at the end
+          of the negotiation. The Token Binding Id can then simply be looked up whenever needed.
+          If the User Agent is unable to negotiate Token Binding with a server, it will not create
+          a Token Binding Id. If the Token Binding Id for a server is deleted by a User Agent
+          (e.g., as a result of some user action), the User Agent will need to re-establish a TLS
+          connection to negotiate Token Binding and create a new Token Binding Id.
+          <li><p>If <var>providedTokenBindingId</var> is not <var>null</var>, then add a
+            <code title="">provided_token_binding</code> to <var>tokenBindingMessage</var>.
+          <li><p>If the <var>httpRequest</var>'s <var>use-referred-token-binding</var> flag is true,
+            then run these substeps:
+          <ol>
+            <li><p>Let <var>referredTokenBindingId</var> be the result of looking up the Token
+              Binding Id for the <var>httpRequest</var>'s <var>referred-token-binding-origin</var>.
+              <p class="note no-backref">A server indicates that it want to use a Referred Token
+              Binding when doing a redirect to another server, or when code from that server's
+              origin executes in the User Agent.  The User Agent would have negotiated the use of
+              Token Binding with that server when it set up the TLS connection to that server,
+              creating a Token Binding Id at the end of the negotiation. This Token Binding Id can
+              then simply be looked up whenever needed. If the Token Binding Id for the server is
+              deleted by a User Agent (e.g., as a result of some user action), the User Agent will
+              need to re-establish a TLS connection to negotiate Token Binding and create a new
+              Token Binding Id.
+            <li><p>If <var>referredTokenBindingId</var> is different from
+              <var>providedTokenBindingId</var>, then add <var>referredTokenBindingId</var> as a
+              <code title="">referred_token_binding</code> to <var>tokenBindingMessage</var>.
+          </ol>
+
+          <li><p>If <var>tokenBindingMessage</var> is not <var>null</var>, append a
+            <code title="">Sec-Token-Binding</code> header to <var>httpRequest</var>'s
+            <span title=concept-request-header-list>header list</span>.
+            <p>The value of the <code title="">Sec-Token-Binding</code> header is a
+            <var>Token Binding Message</var> as specified in
+            <a href="https://tools.ietf.org/id/draft-ietf-tokbind-https-03.xml#rfc.section.3">section 2</a>
+            of the Token Binding over HTTP spec.
+            The value of a Token Binding Message is specified in
+            <a href="https://tools.ietf.org/id/draft-ietf-tokbind-protocol-06.xml#rfc.section.3">section 3</a>
+            of the Token Binding Protocol spec.
```

Comment:
This uses various different ways to refer to "token binding message". Are they all supposed to be the same variable?